subsys/net/ip/l2/arp.c

```diff
@@ -222,6 +222,10 @@ struct net_pkt *net_arp_prepare(struct net_pkt *pkt)
 	if (!net_if_ipv4_addr_mask_cmp(net_pkt_iface(pkt),
 				       &NET_IPV4_HDR(pkt)->dst)) {
 		addr = &net_pkt_iface(pkt)->ipv4.gw;
+		if (IS_ENABLED(CONFIG_NET_LOG)
```

Comment:
This is wrong, you should check against "CONFIG_NET_DEBUG_ARP" here.

Reply:
Fixed